api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -78,21 +94,69 @@ public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
     this.organizationTypeOtherText = organizationTypeOtherText;
   }
 
+  /** Builder method to help streamline the building of a DbInstitution. */
+  public DbInstitution organizationTypeOtherText(String organizationTypeOtherText) {
+    this.organizationTypeOtherText = organizationTypeOtherText;
+    return this;
+  }
+
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
   public Set<DbInstitutionEmailDomain> getEmailDomains() {
     return emailDomains;
   }
 
-  public void setEmailDomains(Iterable<DbInstitutionEmailDomain> emailDomains) {
-    this.emailDomains =
-        StreamSupport.stream(emailDomains.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First, call
+   * retainAll() to subset DB rows to those we wish to keep: the intersection of old and new. Then,
+   * call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailDomains the new set of domains for this Institution
+   */
+  public void setEmailDomains(final Set<DbInstitutionEmailDomain> emailDomains) {
+    final Set<DbInstitutionEmailDomain> attachedDomains =
+        emailDomains.stream()
+            .map(address -> address.setInstitution(this))
```

Comment:
Need to make the connection in both directions, but we only need to break it in one.  It is NOT necessary to make this null when removing.

Reply:
nit: why are you calling the map variable `address` instead of `ed` or similar? I didn't think domains were addresses.

Reply:
copypasta.  good catch.